### ISSUE TITLE
 False positive in Lint/ShadowedArgument cop

### ISSUE THREAD

**Issue by mikecmpbll** [bbatsov/rubocop#5561]
The ShadowedArgument cop is triggering where the variable may not be being modified at all, the smallest reproducable example I can get is:

```ruby
def resource(opts = nil)
  opts = type if foo
  opts ||= {}

  # ...
end
```

Here, if the `resource` method is called with an argument and `foo` is falsey, opts is not being replaced or "shadowed" in this method.

--------

## Expected behavior

Should pass.

## Actual behavior

```
rubocop test.rb 
Inspecting 1 file
W

Offenses:

test.rb:1:14: W: Lint/ShadowedArgument: Argument opts was shadowed by a local variable before it was used.
def resource(opts = nil)
             ^^^^^^^^^^

1 file inspected, 1 offense detected
```

## Steps to reproduce the problem

See example above.

## RuboCop version

0.52.1 (using Parser 2.4.0.2, running on ruby 2.3.1 x86_64-darwin14)

**Comment by mikegee** [comment]
I confirmed the report, including the fact that both of those conditional lines are required to trigger the offense (neither one is sufficient). This implies the cop tries to avoid false reports when the shadowing is conditional, but this example method is more complicated than it can currently handle.

**Comment by alberto** [comment]
Same issue here
